model/task/db.go

```diff
@@ -2640,12 +2640,14 @@ func getTasksByVersionPipeline(versionID string, opts GetTasksByVersionOptions)
 // fetching tasks to block them, and for each task, all dependencies
 // that have a status that that would block the task from running (i.e., it is
 // inconsistent with the task's Dependency.Status field) and have not been marked
-// unattainable will be retrieved.
+// unattainable will be retrieved. The updateAllDependencies indicates whether
+// all tasks that depend on the given tasks will be updated (i.e., whether
+// the task's Dependency.Status will be ignored).
```

Comment:
should we update 'the task's Dep..' to the 'the blocked task's Dep..' for clarity?

Reply:
I think it should stay as is because this function can be used by both `UpdateUnblockedDependencies` and `UpdateBlockedDependencies`, so the tasks in question can be blocked or unblocked.

Reply:
I think it could help to clarify why we would want to ignore dependency's status since it's not that clear just from inspection. Could we add a motivating example of a situation (i.e. possibly this one) where the desired behavior would be to block all dependencies, even if the dependency status matched the current task status?